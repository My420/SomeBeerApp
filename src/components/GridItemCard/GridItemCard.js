import React from 'react';
import PropTypes from 'prop-types';
import styles from './GridItemCard.module.scss';
import GridCardControl from './GridCardControl/GridCardControl';
import GridCardLink from './GridCardLink/GridCardLink';

class GridItemCard extends React.Component {
  state = { isDescriptionOpen: false };

  showDescription = () => {
    this.setState({ isDescriptionOpen: true });
  };

  hideDescription = () => {
    this.setState({ isDescriptionOpen: false });
  };

  toggleDescription = () => {
    const { isDescriptionOpen } = this.state;
    this.setState({ isDescriptionOpen: !isDescriptionOpen });
  };

  addEndListener = (node, done) =>
    node.addEventListener('transitionend', done, false);

  render() {
    const { data } = this.props;
    const { isDescriptionOpen } = this.state;

    return (
      <div className={styles.card}>
        <GridCardLink
          data={data}
          isDescriptionOpen={isDescriptionOpen}
          onCursorEnter={this.showDescription}
          onCursorLeave={this.hideDescription}
        />
        <GridCardControl
          data={data}
          isDescriptionOpen={isDescriptionOpen}
          onToggleButtonClick={this.toggleDescription}
        />
      </div>
    );
  }
}

GridItemCard.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  data: PropTypes.object.isRequired
};

export default GridItemCard;
