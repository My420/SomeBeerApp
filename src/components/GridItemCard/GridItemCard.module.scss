@import '../../sass/variables.scss';

.card {
  font-size: 18px;
  line-height: 24px;
  border: 2px solid $menu-bar-color;
}

.card:hover {
  box-shadow: 0 0 5px 1px $menu-color;
}
