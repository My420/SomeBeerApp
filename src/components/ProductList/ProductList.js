import React from 'react';
import PropTypes from 'prop-types';
import Immutable from 'immutable';
import styles from './ProductList.module.scss';
import ListGroup from '../ListGroup/ListGroup';
import ListGroupItem from '../ListGroupItem/ListGroupItem';

const ProductList = ({ data, ProductCard }) => {
  // eslint-disable-next-line no-console
  console.log('render ===== ProductList');
  const goods = data.toJS().map(elem => {
    return (
      <ListGroupItem className={styles.item} key={elem.id}>
        <ProductCard data={elem} />
      </ListGroupItem>
    );
  });
  return <ListGroup className={styles.list}>{goods}</ListGroup>;
};

export default ProductList;

ProductList.propTypes = {
  ProductCard: PropTypes.elementType.isRequired,
  data: PropTypes.instanceOf(Immutable.List).isRequired
};
