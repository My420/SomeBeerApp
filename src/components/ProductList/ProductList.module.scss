@import '../../sass/variables.scss';

.list {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;

  list-style: none;
}

.item {
  margin-bottom: 25px;
}

@media (max-width: $MPC-container-width) {
  .list {
    width: 572px;
    margin: 0 auto;
  }
}

@media (max-width: $MPC-container-width) {
  .list {
    width: 572px;
    margin: 0 auto;
  }
}

@media (max-width: 610px) {
  .list {
    width: 266px;
    margin: 0 auto;
  }
}
