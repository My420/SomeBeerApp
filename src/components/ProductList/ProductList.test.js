import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';
import ProductList from './ProductList';
import GridItemCard from '../GridItemCard/GridItemCard';

describe('Test <ProductList /> component', () => {
  const testData = List([
    {
      id: '1'
    },
    {
      id: '2'
    }
  ]);
  const wrapper = shallow(
    <ProductList data={testData} ProductCard={GridItemCard} />
  );

  test('should correct render <ProductList /> component ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <ListGroup/> component with className = list', () => {
    expect(wrapper.find('ListGroup').prop('className')).toBe('list');
  });

  test('should render 2 <ListGroupItem/> component inside <ListGroup/> component', () => {
    expect(wrapper.find('ListGroup').find('ListGroupItem')).toHaveLength(2);
  });

  test('<ListGroupItem/> should have className = item', () => {
    expect(
      wrapper
        .find('ListGroupItem')
        .first()
        .prop('className')
    ).toBe('item');
    expect(
      wrapper
        .find('ListGroupItem')
        .last()
        .prop('className')
    ).toBe('item');
  });

  test('<ListGroupItem/> should have prop key = id', () => {
    expect(
      wrapper
        .find('ListGroupItem')
        .first()
        .key()
    ).toBe(testData.get(0).id);
    expect(
      wrapper
        .find('ListGroupItem')
        .last()
        .key()
    ).toBe(testData.get(1).id);
  });

  test('should render 2 <GridItemCard/> component inside <ListGroupItem/> component', () => {
    expect(
      wrapper
        .find('ListGroupItem')
        .first()
        .find('GridItemCard')
    ).toHaveLength(1);
    expect(
      wrapper
        .find('ListGroupItem')
        .last()
        .find('GridItemCard')
    ).toHaveLength(1);
  });

  test('<GridItemCard/> should have correct prop data', () => {
    expect(
      wrapper
        .find('GridItemCard')
        .first()
        .prop('data')
    ).toEqual(testData.get(0));
    expect(
      wrapper
        .find('GridItemCard')
        .last()
        .prop('data')
    ).toEqual(testData.get(1));
  });
});
