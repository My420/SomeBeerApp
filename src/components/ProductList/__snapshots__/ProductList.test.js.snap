// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ProductList /> component should correct render <ProductList /> component  1`] = `
<ListGroup
  className="list"
  tag="ul"
>
  <ListGroupItem
    className="item"
    key="1"
  >
    <GridItemCard
      data={
        Object {
          "id": "1",
        }
      }
    />
  </ListGroupItem>
  <ListGroupItem
    className="item"
    key="2"
  >
    <GridItemCard
      data={
        Object {
          "id": "2",
        }
      }
    />
  </ListGroupItem>
</ListGroup>
`;
